File: 1669-merge-in-between-linked-lists/1669-merge-in-between-linked-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeInBetween(ListNode list1, int a, int b, ListNode list2) {
        
        ListNode temp = list1;
        ListNode prev = null;
        b = b-a;
        while(a-- > 0){
            prev= temp;
            temp = temp.next;
        }
        
        while(b-- > 0){
            temp = temp.next;
        }
        ListNode end = list2;
        while(end.next!=null){
            end = end.next;
        }
        prev.next = list2;
        end.next = temp.next;
        
        return list1;
        
    }
}

File: 1669-merge-in-between-linked-lists/1669-merge-in-between-linked-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeInBetween(ListNode* list1, int a, int b, ListNode* list2) {
        ListNode* ll = list2;
        while(ll->next!=NULL)ll = ll->next;
        int count = 0;
        ListNode* temp = list1;
        while(count != a-1){
            temp = temp->next;
            count++;
        }
        ListNode* move = temp->next;
        temp->next = list2;
        while(count!=b){move=move->next;count++;}
        ll->next = move;
        return list1;
        
        
        
    }
};